### PR TITLE
Word prop: Define the induced relations, prove some of their properties

### PR DESCRIPTION
**Pull request by gallais**
re #727

### CODE REVIEW

File: src/Data/Word/Properties.agda [hunk 1]
```diff
@@ -47,7 +96,11 @@ _==_ : Word64 → Word64 → Bool
...
--- A strict total ordering over words
+-- Properties of _<_
+
+infix 4 _<?_
+_<?_ : Decidable _<_
+_<?_ = On.decidable toℕ ℕ._<_ ℕₚ._<?_
 
-<-strictTotalOrder : StrictTotalOrder _ _ _
-<-strictTotalOrder = On.strictTotalOrder ℕₚ.<-strictTotalOrder toℕ
+≈-<-strictTotalOrder : StrictTotalOrder _ _ _
```
Comment (MatthewDaggitt) [inline src/Data/Word/Properties.agda]: I think the convention in `Data.Nat.Properties` has been to add the equality to the end of the type, i.e. `<-strictTotalOrder-≈`

Comment (gallais) [reply src/Data/Word/Properties.agda]: Aren't all the strict total orders in `Data.Nat.Properties` based on
propositional equality? I had a quick look at lexicographic ordering
on lists and there we don't mention the underlying equality (the
pointwise lifting of the setoid parameter's) at all!

https://agda.github.io/agda-stdlib/Data.List.Relation.Binary.Lex.Strict.html#3688

I think it makes sense to write:

```agda
≈-<-isStrictTotalOrder : IsStrictTotalOrder _≈_ _<_*
```

and thus to keep the same ordering in all the names (I'm happy we drop
the `≈-` when the equality happens to be propositional equality).

Comment (MatthewDaggitt) [reply src/Data/Word/Properties.agda]: Ah sorry I should have been more specific, I meant things like:
```
+-cancelˡ-≡ : LeftCancellative _≡_ _+_
+-cancelˡ-≤ : LeftCancellative _≤_ _+_
```
I can see the logic of keeping the ordering the same. But I guess my argument would be that the equality is the least important part of the proof. The point of the proof is that we're proving that `_<_` is a strict total order, and it just so happens that it holds over several different equalities. It therefore feels a little strange in my opinion to have the equality come first in the name of the proof.

Comment (gallais) [reply src/Data/Word/Properties.agda]: Ok, that makes sense. Thanks for clarifying!

**Comment by gallais** [comment]
I've removed milestone v1.1 because we don't know whether we'll get a released 
version of Agda with the right word primitive before we reach that milestone.

**Pull request merged**